### ISSUE TITLE
Don't set backend_url in Datatrust.register

### ISSUE THREAD

**Issue by rbharath** [computablelabs/goest#141]
At present the `Datatrust.register` method sets the `backend_url` in https://github.com/computablelabs/goest/blob/master/contracts/Datatrust.vy#L172:
```
self.backend_url = url # we'll clear this if the registration fails
```
This setting was put in place because `resolveRegistration` doesn't have access to the `backend_url` otherwise. However, since `register` can be called by any Ethereum user, it would be possible to run a phishing attack by changing `backend_url` and capturing incoming traffic.

The fix for this is to remove this setter line and have the datatrust operator explicitly call `setBackendUrl` to set their url after their backend registration is accepted by the market.

**Comment by rbharath** [comment]
As an addendum, `Datatrust.resolveRegistration` currently is:
```
@public
def resolveRegistration(hash: bytes32):
  """
  @notice Set internal backend in use if approved (remove if not)
  @param hash The hashed string url of the backend candidate
  """
  assert self.voting.candidateIs(hash, REGISTRATION)
  assert self.voting.pollClosed(hash)
  owner: address = self.voting.getCandidateOwner(hash)
  # case: listing accepted
  if self.voting.didPass(hash, self.parameterizer.getPlurality()):
    self.backend_address = owner
    log.RegistrationSucceeded(hash, owner)
  else: # application did not pass vote
    log.RegistrationFailed(hash, owner)
    clear(self.backend_url)
  # regardless, the candidate is pruned
  self.voting.removeCandidate(hash)
```
With the removal of the line from `Datatrust.register` setting `backend_url`, the `clear` call should be moved to the other branch of the if/else. This is because the datatrust operator is now responsible for setting `backend_url` by calling `Datatrust.setBackendUrl`. So open a successful registration, the old `backend_url` should be reset:
```
@public
def resolveRegistration(hash: bytes32):
  """ 
  @notice Set internal backend in use if approved (remove if not)
  @param hash The hashed string url of the backend candidate
  """
  assert self.voting.candidateIs(hash, REGISTRATION)
  assert self.voting.pollClosed(hash)
  owner: address = self.voting.getCandidateOwner(hash)
  # case: listing accepted
  if self.voting.didPass(hash, self.parameterizer.getPlurality()):
    clear(self.backend_url)                                                                                                                              
    self.backend_address = owner
    log.RegistrationSucceeded(hash, owner)
  else: # application did not pass vote
    log.RegistrationFailed(hash, owner)
  # regardless, the candidate is pruned
  self.voting.removeCandidate(hash)
```

**Comment by rbharath** [comment]
Issue addressed by #147

**Comment by rbharath** [comment]
Re-opening after rollback

**Comment by rbharath** [comment]
Re-implemented by #170
